edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemInfo.cs

```diff
@@ -2,23 +2,25 @@
 
 namespace Microsoft.Azure.Devices.Edge.Agent.Core
 {
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+
+    using static System.Net.WebUtility;
+
     public class SystemInfo
     {
-        public SystemInfo(string operatingSystemType, string architecture, string version, ProvisioningInfo provisioning, string serverVersion, string kernelVersion, string operatingSystem, int numCpus, string virtualized)
+        public SystemInfo(string operatingSystemType, string architecture, string version, IDictionary<string, object> additionalProperties)
```

Comment:
Can you simply use Option<String> instead of object? or even string?

Reply:
The refactor of SystemInfo will come in a separate PR.

Reply:
In that case, I'll add an additional comment to make this an IReadOnlyDictionary

Reply:
Done here: dc364951af1a9db9f1a9a5f8843c4f829ae7bfe3. Rust-land has spoiled me with default-read-only collections.